### PR TITLE
Edge validity

### PR DESCRIPTION
**Pull request by bsaund**
@dmcconachie I think we are the only people that use this.

This adds an explicit "edge validity" parameter to each edge. 
This adds a byte to GraphEdge, and therefore changes serialization, so old saved graphs will be broken

Note: Validity could be implicitly accomplished, using -1 for unknown, and inf for invalid or something

### CODE REVIEW

**Comment by dmcconachie** [comment]
I'm going to put off reviewing this until after the IROS deadline unless there is some particular reason to look at it now.

**Comment by bsaund** [comment]
There's no rush, just means I'm not using master branch of arc utilities

**Comment by calderpg** [comment]
We have a similar use case for adding additional information to graph edges - in our case tags and indices for global roadmap updates via GPU - that is part of [common_robotics_utilities'](https://github.com/calderpg/common_robotics_utilities/blob/master/include/common_robotics_utilities/simple_graph.hpp) graph class. Our solution was a `uint64_t` per edge which is almost "free" given that it bumps the object size up to the next multiple of 16-bytes stack alignment and allows both integer and bitmask values to be stored.

File: include/arc_utilities/dijkstras.hpp [hunk 1]
```diff
@@ -388,9 +423,10 @@ namespace arc_dijkstras
         Graph()
         {}
 
+
         uint64_t SerializeSelf(
-                std::vector<uint8_t>& buffer,
-                const std::function<uint64_t(const NodeValueType&, std::vector<uint8_t>&)>& value_serializer) const
+            std::vector<uint8_t>& buffer,
```
Comment (dmcconachie) [inline include/arc_utilities/dijkstras.hpp]: These whitespace changes appear to make these 2 functions have different formatting from the rest of the file. Any particular reason for the change?

Comment (bsaund) [reply include/arc_utilities/dijkstras.hpp]: Good catch
Editor preferences. I will return to original indentation

**Comment by bsaund** [comment]
FYI, I have other unmerged changes related to arc_dijkstras that are not yet ready for a PR.
Will address this soon

File: include/arc_utilities/dijkstras.hpp (hunk 1 above)
Comment (dmcconachie) [reply include/arc_utilities/dijkstras.hpp]: Ping.

**Pull request merged**